client/keys/add.go

```diff
@@ -75,6 +76,23 @@ the flag --nosort is set.
 	return cmd
 }
 
+func getKeybase(cmd *cobra.Command, dryrun bool) (keys.Keybase, error) {
+	if dryrun {
+		return keys.NewInMemory(), nil
+	}
+
+	return NewKeyringFromHomeFlag(cmd.InOrStdin())
```

Comment:
This may cause issues. This function should accept a buffer as an input cause code in `RunAddCmd` may allocate a buffer with `cmd.InOrStdin()` as input stream (which is what it does in fact)

Reply:
Let me know if I have to reset the buffer or do anything else (not exactly sure how this is used)